### DISCUSSION TITLE
Adblocker blocks self-hosted analytics

### DISCUSSION THREAD

**Discussion by alkarkhi** [plausible/analytics#965] [Self-Hosted Support] [answered]
I self-hosted plausible in my own domain at plausible.domain.tld and I realised ad-blockers are blocking it. I tested with and without ad-blockers (uBlock Origin and Adguard and other custom ones bundled with browsers like Brave, Opera), and when I tested with adblockers I got no results whereas when I turned them off I started to get results. i've heard If I self hosted it wouldn't be a problem? Or is it? Am I doing something wrong?

**Comment by metmarkosaric** [comment]
i think it's best to discuss this with the relevant blocklist as we have no control over what they do

**Comment by alkarkhi** [comment]
Is that normal behaviour though? As far as I'm aware, my domain isn't in any blocklist.

**Reply by metmarkosaric** [reply]
yes, it's normal for adblockers to block self-hosted analytics too. for them in general there's no difference what you use, they try to block it all

**Reply by alkarkhi** [reply]
From this issue, https://github.com/plausible/analytics/issues/158 I've seen self-hosted being recommended to avoid getting blocked. I guess they started to use regex or something to block anything with "plausible" or "analytics" in it

**Reply by metmarkosaric** [reply]
you can check the specific blocklist to see why they blocked it on your site but yeah anything with stats, analytics, plausible, matomo etc in the file name or domain name could trigger the block regardless of being self-hosted or not

**Reply by flashblaze** [reply]
Interesting. I haven't run into this issue. I'm self hosting plausible on https://stats.notesbin.xyz and using it on https://flashblaze.xyz and https://notesbin.xyz. And I'm not getting blocked.

You can see my public stats on https://stats.notesbin.xyz/flashblaze.xyz. Both of my sites are on my GitHub profile. Feel free to reach out and inspect.

**Reply by alkarkhi** [reply]
> 
> 
> Interesting. I haven't run into this issue. I'm self hosting plausible on https://stats.notesbin.xyz and using it on https://flashblaze.xyz and https://notesbin.xyz. And I'm not getting blocked.
> 
> You can see my public stats on https://stats.notesbin.xyz/flashblaze.xyz. Both of my sites are on my GitHub profile. Feel free to reach out and inspect.

I think it's blocked too. Didn't seem to register a unique visitor in firefox with ublock origin

**Reply by flashblaze** [reply]
@DodyAlkarkhi I'm running uBlock too, but I was registered as a user right now when I tried it.

Edit: If you visited this https://notesbin.xyz/ then it logged you as a user. I haven't made those stats public btw.

**Reply by alkarkhi** [reply]
I just turned off ublock origin for your site and it still didn't register a click. I also turned off noscript and firefox strict protections. That's weird.

Edit; Where does it say I'm from?

**Reply by flashblaze** [reply]
It says you're from UK. I've attached imgur link for the screenshots

https://imgur.com/a/Tfc7G6y

**Reply by alkarkhi** [reply]
hmm.. That's weird. I guess it's the name of the subdomain then. Although I'd expect adblockers to block "stats" too. Do you mind trying my site?

**Reply by flashblaze** [reply]
Can you tell me what your site is?

**Reply by alkarkhi** [reply]
https://www.alkarkhi.com/

**Reply by flashblaze** [reply]
Yeah, it is blocked
![image](https://user-images.githubusercontent.com/13968549/116778223-3d922d00-aa8e-11eb-9b4d-e3b6a8a3d8eb.png)

**Reply by alkarkhi** [reply]
Did you change the name of the script? For some reason I can't see it in the source.

**Reply by alkarkhi** [reply]
Because I looked at the easyprivacy list and I see they block anything /plausible.js, which is in my source code. How do you change the name of the script @metmarkosaric

**Reply by flashblaze** [reply]
I haven't changed anything. Here is how I'm loading my script.

```
        <script
          async
          defer
          data-domain="notesbin.xyz"
          src="https://notesbin.xyz/js/index.js"></script>
```

So the Plausible script is a folder which has all the static content where I've put my script and then I'm just referencing/loading it.

**Reply by alkarkhi** [reply]
Mine came out as https://plausible.alkarkhi.com/js/plausible.js. How did you get it as index.js?
What I think ublock origin is blocking is the part at the end: /plausible.js

**Reply by alkarkhi** [reply]
@flashblaze Yep, that was it. I used the comment from @hyperknot at https://github.com/plausible/analytics/discussions/387 to change plausible.js to index.js and now it working with ublock origin. I think changing the filename is something that should be an option @ukutaht @metmarkosaric

**Reply by flashblaze** [reply]
Could you mark this conversation as answered if it is working for you, as it might help others too.

**Comment by markcellus** [comment]
Yeah it sucks. Does anyone know if tracking server side would have the same result?

**Reply by metmarkosaric** [reply]
If you're concerned about adblockers, we have a custom domain functionality that you can enable on the cloud https://plausible.io/docs/custom-domain

With the custom domain enabled, Chrome and Safari adblockers will not block Plausible. Only uBlock Origin when used with Firefox can still block it and Brave browser itself. We plan to fix those cases in the future too but they stand for a small percentage of traffic

alternative is to proxy our script. see here how https://github.com/plausible/analytics/discussions/610#discussioncomment-291102

**Comment by alkarkhi** [comment, marked as answer]
I used the comment from @hyperknot at #387 to change plausible.js to index.js (You can change to anything) and now it working with ublock origin.

**Comment by metmarkosaric** [comment]
Leaving this here in case someone finds this thread using search:

If you're concerned about adblockers on the cloud version, we now have instructions on how to proxy our script. See https://plausible.io/docs/proxy/introduction

**Reply by vectorselector** [reply]
In case one is having difficulty proxying {site}/js/plausible.js  it is always possibly, albeit "hacky", to host plausible.js locally on your webserver as whatever.js IF you alter the callback URL

Since plausible.js is only the equivalent of a handful of lines of unminified JS, edit it and alter: 

`s=o.getAttribute("data-api")||new URL(o.src).origin+"/api/event"`
to read
`s="https://plausible.mydomain.com/api/event"`

Of course, if plausible.js gets updated, you need to replace your file, but as I said, it's very little code to maintain if necessary to do so.
Plausible is so simple and this makes it brilliant!
